xonsh/built_ins.py

```diff
@@ -949,6 +949,10 @@ def run_subproc(cmds, captured=False, envs=None):
     if builtins.__xonsh__.env.get("XONSH_TRACE_SUBPROC"):
         print("TRACE SUBPROC: %s" % str(cmds), file=sys.stderr)
 
+    precmds = events.on_subproc_precommand.fire(cmds=cmds)
```

Comment:
Maybe this should be called `change_subproc_cmds`?

Reply:
Or maybe "on_run_subproc()"

Reply:
Please tell me the name you want and I change :)

Reply:
`on_run_subproc()` is best I think